### PR TITLE
Implement java.util.AbstractCollection.addAll.

### PR DESCRIPTION
**Pull request by nicolasstucki**

### CODE REVIEW

**Comment by nicolasstucki** [comment]
Review by @sjrd

**Comment by sjrd** [comment]
Test?

**Comment by nicolasstucki** [comment]
There, forgot to copy the tests to this branch.

File: test-suite/src/test/scala/org/scalajs/testsuite/javalib/AbstractCollectionTest.scala [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  }
+}
+
+class AbstractCollectionFactory extends CollectionFactory {
+
+  override def implementationName: String =
+    "java.util.AbstractCollection"
+
+  override def empty[E]: ju.AbstractCollection[E] = {
+    // inefficient but simple for debugging implementation of AbstractList
```
Comment (sjrd) [inline test-suite/src/test/scala/org/scalajs/testsuite/javalib/AbstractCollectionTest.scala]: The comment says AbstractList. It should say AbstractCollection.

File: test-suite/src/test/scala/org/scalajs/testsuite/javalib/AbstractCollectionTest.scala [hunk 2]
```diff
@@ -0,0 +1,70 @@
+package org.scalajs.testsuite.javalib
+
+import java.{util => ju}
+
+object AbstractCollectionTest extends AbstractCollectionTest(new AbstractCollectionFactory)
+
+abstract class AbstractCollectionTest[F <: AbstractCollectionFactory](val factory: F)
+  extends CollectionTest[F] {
```
Comment (sjrd) [inline test-suite/src/test/scala/org/scalajs/testsuite/javalib/AbstractCollectionTest.scala]: 4 spaces of indentation.

**Comment by sjrd** [comment]
That's all.

**Comment by sjrd** [comment]
LGTM

**Pull request merged**